packages/web-components/src/radio/radio.spec.ts

```diff
@@ -205,82 +165,87 @@ test.describe('Radio', () => {
     });
 
-    test('should set its checked property to true if the checked attribute is set', async () => {
-      await root.evaluate(node => {
-        node.innerHTML = /* html */ `
-                    <form>
-                        <fluent-radio checked></fluent-radio>
-                    </form>
-                `;
-      });
-
+    test('should set its `checked` property to true if the `checked` attribute is set', async ({ page }) => {
+      const element = page.locator('fluent-radio');
       const form = page.locator('form');
 
-      expect(await element.evaluate(node => node.hasAttribute('checked'))).toBeTruthy();
+      await page.setContent(/* html */ `
+          <form>
+              <fluent-radio checked></fluent-radio>
+          </form>
+      `);
 
-      await expect(element).toBeChecked();
+      await expect(element).toHaveAttribute('checked');
+
+      await expect(element).toHaveJSProperty('checked', true);
 
       await element.evaluate((node: Radio) => (node.checked = false));
 
-      await expect(element).not.toBeChecked();
+      await expect(element).toHaveAttribute('checked');
+
+      await expect(element).toHaveJSProperty('checked', false);
 
       await form.evaluate((node: HTMLFormElement) => {
         node.reset();
       });
 
-      await expect(element).toBeChecked();
-    });
+      await expect(element).toHaveAttribute('checked');
 
-    test('should put the control into a clean state, where `checked` attribute modifications modify the `checked` property prior to user or programmatic interaction', async () => {
-      await root.evaluate(node => {
-        node.innerHTML = /* html */ `
-                    <form>
-                        <fluent-radio>Radio</fluent-radio>
-                    </form>
-                `;
-      });
+      await expect(element).toHaveJSProperty('checked', true);
+    });
 
+    test('should put the control into a clean state, where `checked` attribute modifications modify the `checked` property prior to user or programmatic interaction', async ({
+      page,
+    }) => {
+      const element = page.locator('fluent-radio');
       const form = page.locator('form');
 
+      await page.setContent(/* html */ `
+          <form>
+              <fluent-radio>Radio</fluent-radio>
+          </form>
+      `);
+
       await element.evaluate((node: Radio) => {
         node.checked = true;
       });
 
       await element.evaluate(node => node.removeAttribute('checked'));
 
-      await expect(element).toBeChecked();
+      await expect(element).toHaveJSProperty('checked', true);
 
       await form.evaluate((node: HTMLFormElement) => {
         node.reset();
       });
 
-      await expect(element).not.toBeChecked();
+      await expect(element).toHaveJSProperty('checked', false);
 
       await element.evaluate(node => node.setAttribute('checked', ''));
 
-      await expect(element).toBeChecked();
+      await expect(element).toHaveJSProperty('checked', true);
```

Comment:
I found the easiest way to handle form submission validation is to create a form with `GET` submission method, set the element state, submit the form, and then look at the URL. I'll port some of the Checkbox tests which do that over to Radio.

Reply:
I added some tests to radio-group around form and value submission. The `<fluent-radio>` shouldn't submit its own value with the form, since that's handled by the `<fluent-radio-group>`.